File: infra/src/main/java/com/beat/infra/auth/social/kakao/response/KakaoAccount.java
package com.beat.infra.auth.social.kakao.response;

import com.fasterxml.jackson.annotation.JsonProperty;

public record KakaoAccount(
	@JsonProperty("profile_nickname_needs_agreement")
	Boolean profileNicknameNeedsAgreement,
	@JsonProperty("profile")
	KakaoUserProfile profile,
	@JsonProperty("email_needs_agreement")
	Boolean emailNeedsAgreement,
	@JsonProperty("is_email_valid")
	Boolean emailValid,
	@JsonProperty("is_email_verified")
	Boolean emailVerified,
	@JsonProperty("email")
	String email
) {
}

File: infrastructure/src/main/kotlin/com/beat/infrastructure/external/social/kakao/response/KakaoAccount.kt
package com.beat.infrastructure.external.social.kakao.response

import com.fasterxml.jackson.annotation.JsonProperty

internal data class KakaoAccount(
    @param:JsonProperty("profile_nickname_needs_agreement")
    val profileNicknameNeedsAgreement: Boolean?,
    @param:JsonProperty("profile")
    val profile: KakaoUserProfile?,
    @param:JsonProperty("email_needs_agreement")
    val emailNeedsAgreement: Boolean?,
    @param:JsonProperty("is_email_valid")
    val emailValid: Boolean?,
    @param:JsonProperty("is_email_verified")
    val emailVerified: Boolean?,
    @param:JsonProperty("email")
    val email: String?,
)
